src/gamepad.cpp

```diff
@@ -0,0 +1,374 @@
+#include "gamepad.h"
+#include "debug.h"
+#include "sdl_wrappers.h"
+
+#define dbg(x) DebugLog((x),D_SDL) << __FILE__ << ":" << __LINE__ << ": "
+
+extern input_event last_input;
+
+namespace gamepad
+{
+
+constexpr int max_triggers = 2;
+constexpr int max_sticks = 2;
+constexpr int max_axis = 6;
+constexpr int max_buttons = 30;
+
+static int triggers_axis[max_triggers] = {
+    SDL_CONTROLLER_AXIS_TRIGGERLEFT,
+    SDL_CONTROLLER_AXIS_TRIGGERRIGHT
+};
+static int sticks_axis[max_sticks][2] = {
+    { SDL_CONTROLLER_AXIS_LEFTX,  SDL_CONTROLLER_AXIS_LEFTY  },
+    { SDL_CONTROLLER_AXIS_RIGHTX, SDL_CONTROLLER_AXIS_RIGHTY }
+};
+
+static int triggers_state[max_triggers] = {0, 0};
+static int sticks_state[max_sticks] = {0, 0};
+static int hat_state = 0;
+
+static int triggers_threshold = 16000;
+static int sticks_threshold = 16000;
+static int error_margin = 2000;
+
+static int sticks_map[max_sticks][16] = {0};
+static int triggers_map[max_triggers] = {0};
+static int buttons_map[max_buttons] = {0};
+static int hat_map[16] = {0};
+
+struct task_t {
+    Uint32 when;
+    int button;
+    int counter;
+    int state;
+};
+
+constexpr int max_tasks           = max_buttons + max_sticks + max_triggers + 1;
+constexpr int buttons_task_index  = 0;
```

Comment:
```suggestion
// constexpr int buttons_task_index  = 0;
```

Reply:
even if the variable `button_task_index` is not used directly, it's more like a comment (or documentation) to remind me where those button tasks start in the whole bunch.